lib/core/test/unit/Cardano/Wallet/Primitive/TypesSpec.hs

```diff
@@ -499,3 +514,22 @@ instance Arbitrary WalletName where
 instance Arbitrary BoundType where
     shrink = genericShrink
     arbitrary = genericArbitrary
+
+newtype NonZero a = NonZero a
+
+instance (Arbitrary a, Num a, Eq a) => Arbitrary (NonZero a) where
+    arbitrary = NonZero . restrict <$> arbitrary
+      where
+        restrict x = if x == 0 then 1 else x
+    shrink (NonZero x) =
+        map NonZero
+        . suggestOne
+        . filter (/= 0)
+        $ shrink x
+      where
+        suggestOne shrinks = if shrinks == [] then [1] else []
+
+deriving via UTCTime instance Arbitrary StartTime
+deriving via (NonZero NominalDiffTime) instance Arbitrary SlotLength
+deriving via (NonZero Word64) instance Arbitrary EpochLength
```

Comment:
Couldn't resist trying 😅 

We need https://www.stackage.org/lts-13.30/package/stylish-haskell-0.9.2.2 to get CI to accept this, which I think maybe is straight-forward to get if we just bump LTS?

Reply:
Also, the shrinking might be wrong 🤔

Reply:
> maybe is straight-forward to get if we just bump LTS?

It's never straightforward to bump the LTS :sweat_smile: ... Always bring tons of problems with the CI and caches that get us busy for half a day usually. But, sure, we'll have to eventually.